### DISCUSSION TITLE
Problems performing a query using 'OR'.

### DISCUSSION THREAD

**Discussion by vitorvk2** [ujjwalguptaofficial/JsStore#337] [General]
I noticed that in version 4.5.7 there is a problem where using 'OR' in Where within a relationship does not work, and it is not possible to retrieve the additional results.

Making use of the demo environment https://ujjwalguptaofficial.github.io/idbstudio/ we can place this query:

```
select({
    from: 'Orders',
    join: {
        with: 'Customers',
        on: "Orders.customerId=Customers.customerId",
        where: {
            country: 'Mexico',
            or: {
                city: 'London'
            }
        }
    }
});
```

Note that it will only return results for 'Mexico' and the city of 'London' will not be returned. The interesting thing is that in the common Where (without being inside a JOIN) it works properly.

**Comment by ujjwalguptaofficial** [comment]
I think this is an issue in newer version - i am working on this to fix.

**Reply by vitorvk2** [reply]
Thanks!!

**Comment by ujjwalguptaofficial** [comment]
I have fixed this issue and released a new version - 4.5.8. Please install and let me know. @vitorvk2

**Reply by vitorvk2** [reply]
Yes it worked, thank you very much!

**Comment by vitorvk2** [comment]
Hello everything is fine? @ujjwalguptaofficial 

I want to know if it is possible to perform a JOIN of 1 x N

Where all JOIN elements are inside an array.

Example:
```
const tblCustomerChats = {
	name: 'CustomerChats',
	columns: {
	    id: {
		primaryKey: true,
		autoIncrement: false,
		dataType: DATA_TYPE.Number,
	    },
	    hash: {
		dataType: DATA_TYPE.String,
	    },
	    message: {
		dataType: DATA_TYPE.String,
	    },
	}
}


const tblCustomerChatReactions = {
	name: 'CustomerChatReactions',
	columns: {
	    id: {
		primaryKey: true,
		autoIncrement: false,
		dataType: DATA_TYPE.Number,
	    },
	    reaction_text: {
		dataType: DATA_TYPE.String
	    },
	    customer_chat_id: {
		dataType: DATA_TYPE.Number,
	    }
	}
}
```


return:

```
{
	id : 1,
	hash: "fseFGSEG4",
	message: "Hello"
	reactions: [
		{id: 23, reaction_text: ":)", customer_chat_id: 1}
		{id: 25, reaction_text: ":(", customer_chat_id: 1}
		{id: 50, reaction_text: ">:(", customer_chat_id: 1}
	]
}
```
